adapters/repos/db/shard_backup.go

```diff
@@ -53,6 +52,13 @@ func (s *Shard) HaltForTransfer(ctx context.Context) (err error) {
 	if err = s.cycleCallbacks.geoPropsCombinedCallbacksCtrl.Deactivate(ctx); err != nil {
 		return fmt.Errorf("pause geo props maintenance: %w", err)
 	}
+
+	// pause indexing
+	s.queue.Pause()
```

Comment:
This will panic if target vectors are enabled. L.62 is a good example about how to deal with this correctly.

Something like this would work:

```go
if  s.hasTargetVectors() {
 for _, q := range s.Queues() {
	q.Pause()
  }
} else {
  s.queue.Pause()
}
```

Reply:
aha, good to know, i wasn't sure what is the diff. thanks good call out